jira/client.py

```diff
@@ -310,53 +310,35 @@ def _sort_and_quote_values(self, values):
         return [quote(value, safe="~") for value in ordered_values]
 
 
-class JiraCookieAuth(AuthBase):
-    """Jira Cookie Authentication.
-
-    Allows using cookie authentication as described by `jira api docs <https://developer.atlassian.com/server/jira/platform/cookie-based-authentication/>`_
-    """
+class RetryingJiraAuth(AuthBase):
+    """Base class for Jira authentication handlers that need to retry requests on 401 responses."""
 
-    def __init__(
-        self, session: ResilientSession, session_api_url: str, auth: tuple[str, str]
-    ):
-        """Cookie Based Authentication.
-
-        Args:
-            session (ResilientSession): The Session object to communicate with the API.
-            session_api_url (str): The session api url to use.
-            auth (Tuple[str, str]): The username, password tuple.
-        """
+    def __init__(self, session: ResilientSession | None = None):
```

Comment:
Let's remove the default of `None` here, and then remove as many if not all the plain asserts.

If there do end up being situations where the session is None when passing to the constructor of the subclasses then let us move to a subclass `__init__` method override, and rather than asserts, either use warnings, or if it should really fail, let's use an Exception with suitable message.
I want to avoid `asserts` in the production codebase.